scripts/hotfixes/kdump/create-kdump-artifacts.sh

```diff
@@ -0,0 +1,257 @@
+#!/bin/bash
+#
+# MIT License
+#
+# (C) Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+set -euo pipefail
+
+# Source common dracut parameters.
+. "$(dirname $0)/dracut-lib.sh"
+
+# show the line that we failed and on and exit non-zero
+
+trap 'catch $? $LINENO; cleanup; exit 1' ERR
+
+# if the script is interrupted, run the cleanup function
+trap 'cleanup' INT EXIT
+
+# catch() prints what line the script failed on, runs the cleanup function, and then exits non-zero
+catch() {
+    # Show what line the error occurred on because it can be difficult to detect in a dracut environment
+    echo "CATCH: exit code $1 occurred on line $2 in $(basename "${0}")"
+    cleanup
+    exit 1
+}
+
+
+# cleanup() removes temporary files, puts things back where they belong, etc.
+cleanup() {
+    echo "CLEANUP: cleanup function running ..."
+
+    # Restore the dracut config if it was removed.
+    [ ! -f /etc/dracut.conf.d/05-metal.conf ] && cp -v /run/rootfsbase/etc/dracut.conf.d/05-metal.conf /etc/dracut.conf.d/05-metal.conf
+
+    # Stops re-running this script on reboot
+    # FIXME: Don't we want this to re-run?
```

Comment:
Runs once to create the kdump initrd creation script (this script).  No need to run that again unless a node is reimaged, right?

Reply:
Fixing in https://github.com/Cray-HPE/metal-provision/pull/12 and updating this.